fidesctl/src/fidesapi/crud.py

```diff
@@ -160,17 +183,24 @@ async def update(
         sql_model = sql_model_map[resource_type]
         resource_dict = resource.dict()
         query_result = update_resource(sql_model, resource_dict, fides_key)
+        if not query_result:
+            raise NotFoundError(resource_type, fides_key)
+
         return query_result
 
-    @router.delete("/{fides_key}", status_code=status.HTTP_204_NO_CONTENT)
```

Comment:
what's your thinking in changing this to a 200 with a response? I've seen anecdotally that its common to use 204 for deletes and not return anything

Reply:
There's definitely some room for opinion with any HTTP status code decision. In this case, I think the two philosophies are:
- Return a 200, because the request was valid and completed successfully. Send a response payload to provide feedback to the caller, ideally including the resource that was deleted.
- Return a 204, because the request was valid and completed successfully, thus removing the resource, which obviously couldn't be fetched afterwards to include in a response.

IMO, the first option is the right choice for this API. The request and database query burdens are low currently, and I would expect that to be true even in a production environment. Because we're not attempting to scale this API to handle millions+ of req/s, we can afford to include a response payload that provides more context and feedback. If that changes later, we can easily switch to a 204/zero-payload response schema (although, of course, it would constitute a breaking change).

Reply:
makes sense to me!